### PR TITLE
refactor unit tests for namespace and operator group

### LINKED ISSUE #243: Refactor namespace and operator group unit tests
refactor existing `namespace_tests.go` and `operator_group_tests.go` to split them into multiple unit tests instead of just one function.

### PR DESCRIPTION
**Pull request by yashoza19**
## Description of PR
Fixes #243 

## Checklist (required)
- [x] I have reviewed and followed the [contribution guidelines](https://github.com/opdev/opcap/blob/main/docs/contribution.md)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation if needed
- [x] I have checked that my changes pass all tests

### CODE REVIEW

File: internal/operator/namespace_test.go [hunk 1]
```diff
@@ -12,46 +12,50 @@ import (
...
-			ns, err := operatorClient.CreateNamespace(context.TODO(), "testns")
 			It("should succeed", func() {
+				ns, err := operatorClient.CreateNamespace(context.TODO(), "testns")
 				Expect(err).ToNot(HaveOccurred())
 				Expect(ns).ToNot(BeNil())
 			})
 		})
 		When("creating a namespace that already exists", func() {
 			JustBeforeEach(func() {
+				operatorClient.CreateNamespace(context.TODO(), "testns")
```
Comment (bcrochet) [inline internal/operator/namespace_test.go:35]:
```suggestion
				_, err := operatorClient.CreateNamespace(context.TODO(), "testns")
				Expect(err).ToNot(HaveOccurred())
```

File: internal/operator/operator_group_test.go [hunk 2]
```diff
@@ -13,49 +13,54 @@ import (
 
 var _ = Describe("OperatorGroup", func() {
 	logger.InitLogger("debug")
+	var operatorClient operatorClient
+	var operatorGroupData OperatorGroupData
```
Comment (bcrochet) [inline internal/operator/operator_group_test.go:17]: Same as the namespace. Just leave the vars out of the BeforeEach. Everything else in.

File: internal/operator/operator_group_test.go [hunk 3]
```diff
@@ -13,49 +13,54 @@ import (
...
-			og, err := operatorClient.CreateOperatorGroup(context.TODO(), operatorGroupData, "testns")
 			It("should succeed", func() {
+				og, err := operatorClient.CreateOperatorGroup(context.TODO(), operatorGroupData, "testns")
 				Expect(err).ToNot(HaveOccurred())
 				Expect(og).ToNot(BeNil())
 			})
 		})
 		When("creating an operator group with an existing name", func() {
 			JustBeforeEach(func() {
+				operatorClient.CreateOperatorGroup(context.TODO(), operatorGroupData, "testns")
```
Comment (bcrochet) [inline internal/operator/operator_group_test.go:40]: Ditto

File: internal/operator/namespace_test.go [hunk 4]
```diff
@@ -12,46 +12,50 @@ import (
...
+				Expect(err).To(HaveOccurred())
+				Expect(err).To(MatchError("could not create namespace: testns: namespaces \"testns\" already exists"))
+				Expect(ns).To(BeNil())
 			})
 		})
 	})
 	Context("DeleteNamespace", func() {
 		When("deleting the existing namespace", func() {
-			err := operatorClient.DeleteNamespace(context.TODO(), "testns")
 			It("should succeed", func() {
```
Comment (bcrochet) [inline internal/operator/namespace_test.go:47]: You need to create the namespace here, so you have something to delete.

File: internal/operator/namespace_test.go [hunk 5]
```diff
@@ -12,46 +12,50 @@ import (
...
+				err := operatorClient.DeleteNamespace(context.TODO(), "testns")
 				Expect(err).ToNot(HaveOccurred())
 			})
 		})
 		When("deleting a namespace that does not exist", func() {
 			JustBeforeEach(func() {
+				operatorClient.DeleteNamespace(context.TODO(), "testns")
```
Comment (bcrochet) [inline internal/operator/namespace_test.go:54]: This is unnecessary.

File: internal/operator/namespace_test.go [hunk 6]
```diff
@@ -12,46 +12,50 @@ import (
...
+			})
+			It("should return error", func() {
 				err := operatorClient.DeleteNamespace(context.TODO(), "testns")
-				It("should return error", func() {
-					Expect(err).To(MatchError("could not delete namespace: testns: namespaces \"testns\" not found"))
-				})
+				Expect(err).To(MatchError("could not delete namespace: testns: namespaces \"testns\" not found"))
```
Comment (bcrochet) [inline internal/operator/namespace_test.go:58]: I don't think we care _what_ the error is, just that it happens. Instead, I would just wrap the previous line in 

```
Expect(...).To(HaveOccurred())
```

File: internal/operator/operator_group_test.go (hunk 2 above)
Comment (bcrochet) [reply internal/operator/operator_group_test.go:17]: In fact, pretty much all of my comments there apply here.

**Review by bcrochet — CHANGES_REQUESTED** [review]
A few more suggestions.

**Review by bcrochet — APPROVED** [review]
/lgtm

File: internal/operator/namespace_test.go [hunk 7]
```diff
@@ -12,46 +12,50 @@ import (
...
+			})
 		})
 		When("deleting a namespace that does not exist", func() {
-			JustBeforeEach(func() {
-				err := operatorClient.DeleteNamespace(context.TODO(), "testns")
-				It("should return error", func() {
-					Expect(err).To(MatchError("could not delete namespace: testns: namespaces \"testns\" not found"))
-				})
+			It("should return error", func() {
+				Expect(operatorClient.DeleteNamespace(context.TODO(), "testns")).To(HaveOccurred())
```
Comment (bcrochet) [inline internal/operator/namespace_test.go:58]: Nit: Would read better as ToNot(Succeed())

File: internal/operator/operator_group_test.go [hunk 8]
```diff
@@ -12,50 +12,55 @@ import (
...
 	Context("DeleteOperatorGroup", func() {
 		When("deleting the existing operator group", func() {
-			err := operatorClient.DeleteOperatorGroup(context.TODO(), "testog", "testns")
-			It("should delete the existing operator group", func() {
+			JustBeforeEach(func() {
+				_, err := operatorClient.CreateOperatorGroup(context.TODO(), operatorGroupData, "testns")
 				Expect(err).ToNot(HaveOccurred())
 			})
+			It("should delete the existing operator group", func() {
+				Expect(operatorClient.DeleteOperatorGroup(context.TODO(), "testog", "testns")).ToNot(HaveOccurred())
```
Comment (bcrochet) [inline internal/operator/operator_group_test.go:58]: Ditto.

File: internal/operator/operator_group_test.go [hunk 9]
```diff
@@ -12,50 +12,55 @@ import (
...
+			})
 		})
 		When("deleting an operator group that does not exist", func() {
-			JustBeforeEach(func() {
-				err := operatorClient.DeleteOperatorGroup(context.TODO(), "testog", "testns")
-				It("should throw an error", func() {
-					Expect(err).To(MatchError("could not delete operatorgroup: testog: namespace: testns: operatorgroups.operators.coreos.com \"testog\" not found"))
-				})
+			It("should throw an error", func() {
+				Expect(operatorClient.DeleteOperatorGroup(context.TODO(), "testog", "testns")).To(HaveOccurred())
```
Comment (bcrochet) [inline internal/operator/operator_group_test.go:63]: Ditto

**Review by bcrochet — APPROVED** [review]
/lgtm

**Review by acmenezes — APPROVED** [review]
/approve

**Pull request merged**